src/box/lua/schema.lua

```diff
@@ -3500,12 +3529,29 @@ local function revoke(uid, name, privilege, object_type, object_name, options)
     end
     local privilege_hex = privilege_check(privilege, object_type)
     options = options or {}
+    if options.origin == nil then
+        options.origin = ''
+    elseif type(options.origin) ~= 'string' or #options.origin == 0 then
+        box.error(box.error.ILLEGAL_PARAMS, "options parameter 'origin' " ..
+                  "should be of type 'string' and should be non-empty string")
+    end
     local oid = object_resolve(object_type, object_name)
     local _priv = box.space[box.schema.PRIV_ID]
     local _vpriv = box.space[box.schema.VPRIV_ID]
     local tuple = _vpriv:get{uid, object_type, oid}
     -- system privileges of admin and guest can't be revoked
+
+    local origins
     if tuple == nil then
+        origins = {[''] = 0}
+    elseif tuple[6] == nil or tuple[6].origins == nil then
```

Comment:
I vote for using a constant like `PRIV_OPTS_FIELD_ID = 6` constant or simething of this kind.

I also think that it is important to leave a comment that a field id is used instead of a field name deliberately: we don't want to require a database schema upgrade to use the `origin` option.

Reply:
Thanks. fixed.

Reply:
> I also think that it is important to leave a comment that a field id is used instead of a field name deliberately: we don't want to require a database schema upgrade to use the `origin` option.

How about this?

Reply:
Fixed.